src/coreclr/jit/lclmorph.cpp

```diff
@@ -1565,40 +1565,42 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
         // handled in fgCanFastTailCall and fgMakeOutgoingStructArgCopy.
         //
         // CALL(OBJ(LCL_VAR_ADDR...))
-        bool isArgToCall   = false;
-        bool keepSearching = true;
-        for (int i = 0; i < m_ancestors.Height() && keepSearching; i++)
+        // -or-
+        // CALL(LCL_VAR)
+
+        // TODO-1stClassStructs: We've removed most, but not all, cases where OBJ(LCL_VAR_ADDR)
+        // is introduced (it was primarily from impNormStructVal). But until all cases are gone
+        // we still want to handle it as well.
+
+        if (m_ancestors.Height() < 2)
```

Comment:
> UpdateEarlyRefCount needs to recognize LCL_VARs.

Updated that here. I left in the `OBJ(LCL_VAR_ADDR)` handling since we still have that in some cases (even if the majority of `OBJ` wrapping is gone now that it's out of `impNormStrutVal`.

Also changed it from a `for` loop to an explicit to some linear checks instead, which simplifies things and should improve the method TP.

Reply:
-- Not the most familiar with LclMorph, so I went with the assumption that the early exit for `!m_compiler->lvaIsImplicitByRefLocal(lclNum)` is enough to ensure `LCL_VAR` is something we want to handle and that we don't also need to explicitly check that LCL_VAR is a struct or a struct that would actually be passed by ref (e.g. `struct S { int x; }` typically gets passed in register instead).

Let me know if this assumption isn't correct and if more checks are needed.

Reply:
Also did not handle `LCL_FLD` since this is "pre-order" and I believe that means we should only see `GT_FIELD` instead at this point in time.